### PR TITLE
Post server username as metadata

### LINKED ISSUE #1306: post server username as metadata
this is a spin-off from issue #1009: I landed there because I assumed that the username in the credentials would be used as metadata username and I was surprised it kept appearing blank.
What about doing as @lognaturel suggests: in the metadata section, a "use server username" checkbox, defaulting to checked, and if unchecked, then the metadata username would be enabled and used.

However, when it comes to identifying the submitter, I think I don't particularly like the possibility offered here, stating a different identity just by changing a setting, so I guess that in my own forms I will just rely on the IMEI number.

### PR DESCRIPTION
**Pull request by grzesiek2010**
Closes #1306 

#### What has been done to verify that this works as intended?
I've tested implemented changes using this form: 
[preload.txt](https://github.com/opendatakit/collect/files/1245504/preload.txt)

additionally I've tested importing an exporting settings using qr code and simple settings file.

#### Are there any risks to merging this code? If so, what are they?
Can't see any risk here.

### CODE REVIEW

**Comment by lognaturel** [comment]
The functionality is just right and thank you for thinking of the different related functionality to test! I think you got everything. 👍

I have to say I do find the interface confusing. Even though I know what the feature does I managed to be confused when I saw it. That said, it's tricky and I don't have a great solution! Let's see if I can describe what I find confusing and maybe that will give us ideas.

First, there shouldn't be secondary text if it's redundant - https://material.io/guidelines/patterns/settings.html#settings-labels-secondary-text. In fact, maybe just removing that will be enough to make it clear enough.

The other thing that throws me off is that it's not clear to me that the checkbox and the username field after it are related. Maybe flipping the order so that the username goes first would help?

@shobhitagarwal1612 you may also have some good ideas for this UI.

**Comment by grzesiek2010** [comment]
> First, there shouldn't be secondary text if it's redundant

Fixed.

> Maybe flipping the order so that the username goes first would help?

I agree it's not super clear but I don't think so, it's the same order like in `Splash Screen` + `Select splash image` case in `User interface`.

File: collect_app/src/main/res/drawable/settings_color.xml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (yurilaguta) [inline collect_app/src/main/res/drawable/settings_color.xml]: should it be in res/color/ folder instead of drawable? given that it is a colorr selector

**Review by yurilaguta — COMMENTED** [review]
Looks good, except color selector seems to be in wrong folder

File: collect_app/src/main/res/drawable/settings_color.xml (hunk 1 above)
Comment (grzesiek2010) [reply collect_app/src/main/res/drawable/settings_color.xml]: You are right.

File: collect_app/src/main/java/org/odk/collect/android/preferences/ServerPreferencesFragment.java [hunk 2]
```diff
@@ -272,6 +272,7 @@ public boolean onPreferenceChange(Preference preference, Object newValue) {
                 }
 
                 preference.setSummary(username);
+                GeneralSharedPreferences.getInstance().save(KEY_METADATA_USERNAME, username);
```
Comment (lognaturel) [inline collect_app/src/main/java/org/odk/collect/android/preferences/ServerPreferencesFragment.java]: This should only happen if KEY_USE_SERVER_USERNAME is true.

**Review by lognaturel — CHANGES_REQUESTED** [review]

**Comment by lognaturel** [comment]
Thanks for the change, @grzesiek2010 and for your suggestions, @Yurii-Laguta!

One problem -- when "use server username" is unchecked, changes to the server username still affect the form metadata username. They should be entirely independent in that case.

**Comment by grzesiek2010** [comment]
@lognaturel I've fixed those bugs, please review again.

**Review by lognaturel — APPROVED** [review]

**Pull request merged**